### ISSUE TITLE
Python kernel crashes when training new entity type

### ISSUE THREAD

**Issue by CYUlysses** [explosion/spaCy#1476]
Could be a duplicate for issue [#1158](https://github.com/explosion/spaCy/issues/1158), but as the closing comments said this should be solved for 2.0 while I still encounter such issue in 2.0, I am opening a new one.

-----------------------------------------------------------------------------------------------------------------

Hi, I was using a slightly modified version of the provided example code at
https://github.com/explosion/spacy/blob/develop/examples/training/train_new_entity_type.py

The modifications are that I removed the plac components as I am running it within Python environment, and I removed the "disable_pipes" related components as I was prompted with errors saying there's no attribute named disable_pipes for English model.

But every time I only got this output:
----------------Start of output-----------------------------
Created blank 'en' model

Kernel died, restarting
----------------End of output------------------------------

The same problem was not encountered using the "train_ner.py" code in:
https://github.com/explosion/spaCy/blob/develop/examples/training/train_ner.py
(with the same modifications I mentioned above)
and some test runs showed that the loss functions are behaving quite nicely.

However, I do not know how to add multiple different entity types with the train_ner workflow, as I did not find where you can define the name of entity types...?

Please help!

I tried this both on stable 1.9 and 2.0 (with slightly different modifications, but nothing touches the actual training code), and still for the newest subversion. And due to kernel death, I could not find where the traceback was and could not find any crash reports... (unless spaCy keeps some crash log somewhere?)

## Your Environment

* Operating System: Windows 10
* Python Version Used: 3.6
* spaCy Version Used: both 1.9 and 2.0, newest
* Environment Information: IPython console within and without Spyder; also on bare-bone python console.

**Comment by ines** [comment]
> I removed the "disable_pipes" related components as I was prompted with errors saying there's no attribute named disable_pipes for English model.

Sorry about that – as it says in the examples, they were written / updated for `v2.0.0a18`, which is the version currently on `develop` (which will be out today btw).

> However, I do not know how to add multiple different entity types with the train_ner workflow, as I did not find where you can define the name of entity types...?

To add your own entity labels, you can use  `ner.add_label('MY_LABEL')` – just iterate over your labels and add them all. 

I don't remember whether this already worked as expected in `v2.0.0a17` – if not, this might explain a lot and could be the cause of your error. Especially since training worked fine for you using the other NER example. (To debug and verify this, could you add some more intermediate `print` statements between the steps, so we can figure out where exactly it's dying? Right now, it's only clear that it must happen somewhere between loading the model and printing the first loss.)

Once `v2.0.0a18` is out later today, you should also try it again using the new version.

**Comment by ines** [comment]
Update: We've since released v2.0.0 and updated the [`train_new_entity_type.py` example](https://github.com/explosion/spaCy/blob/master/examples/training/train_new_entity_type.py) – plus the [other training examples](https://github.com/explosion/spaCy/blob/master/examples/training/).

Could you try again and see if it works works now? If not, feel free to reopen!

**Comment by ValentinCalomme** [comment]
I have the exact same error. I had it with 1.9, and now that I've upgraded, it still occurs.

Running the `train_ner.py` and `train_new_entity_type.py` files works fine. But somehow, if I try to use my own training data, it crashes.

Could it have something to do with the encoding?

**Comment by ValentinCalomme** [comment]
I seem to have perhaps found the reason why my code crashes, but I don't know how to solve it. After a lot of debugging, it occurred to me that my documents were much larger than the examples provided in `train_ner.py` and `train_new_entity_type.py`

So I tinkered a bit with these examples and replaced one of the documents in the training data by a random string. It seems that there is a glass ceiling on how long the strings are allowed to be without running into problems. It also seems that using a blank model gives you a larger cap on that length (more or less 50000 characters) as opposed to updating a current model (more or less 10000 characters).

@ines could you perhaps investigate this? Maybe there is a solution? Or perhaps this is a strong constraint that we have to deal with?

**Comment by ValentinCalomme** [comment]
@ines the problem still exists...

The example you put online works. But if my training data accounts for more than let's say 3 or 4 new entity types, I cannot pass more than 2 or 3 training instances at a time, otherwise the kernel crashes due to some memory error in the C code running in the background.

Can you guys look into this?

**Comment by CYUlysses** [comment]
@ines I testify that the issue also persists for me.

Also I cannot re-open the issue since it was not closed by me... It would be great if this issue is re-opened, since it is indeed a problem.

**Comment by ValentinCalomme** [comment]
@ines I am now using the CLI and it works. So there must be a difference somewhere there. I use the exact same data and tags and it works using the CLI yet it breaks almost immediately when I try to use it with Python

**Comment by CYUlysses** [comment]
If we cannot re-open the issue here, should I bring this issue to thinc?

**Comment by Hekum** [comment]
Same problem here. Python just crashes when I start training NER. My documents are quite long and contain a lot of entities. Would be great to get some feedback whether this behavior is desired (i.e. are we supposed to feed only sentence-long docs in training?).
